app/lib/verse8/credit.ts

```diff
@@ -0,0 +1,73 @@
+export async function getUserCredit(
+  endpoint: string,
+  userUid: string,
+  creditCredentials: { clientId: string; clientSecret: string },
+): Promise<bigint> {
+  try {
+    const response = await fetch(endpoint + `/v1/users/${userUid}`, {
+      headers: {
+        'Content-Type': 'application/json',
+        'x-client-id': creditCredentials.clientId,
+        'x-client-secret': creditCredentials.clientSecret,
+      },
+    });
+
+    if (response.status === 404) {
+      return 0n;
+    }
+
+    if (!response.ok) {
+      throw new Error('Failed to get user credit');
+    }
+
+    const { credits } = (await response.json()) as {
+      credits: string;
+    };
+
+    return BigInt(credits);
+  } catch (error) {
+    console.error('Failed to get user credit', error);
+    throw new Error('Failed to get user credit');
+  }
+}
+
+export async function consumeUserCredit(
+  endpoint: string,
+  userUid: string,
+  creditCredentials: { clientId: string; clientSecret: string },
+  inputTokens: string,
+  outputTokens: string,
+  description?: string,
+  model: {
+    provider: string;
+    name: string;
+  } = {
+    provider: 'Anthropic',
+    name: 'Claude 3.7 Sonnet',
+  },
```

Comment:
How about restricting `model` as required? it would be tricky part when our standard model changes from sonnet 3.7

Reply:
Agreed 👍